common/config.go

```diff
@@ -0,0 +1,176 @@
+// Copyright © 2023 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/twmb/franz-go/pkg/kgo"
+	"github.com/twmb/franz-go/pkg/sasl"
+	"github.com/twmb/franz-go/pkg/sasl/plain"
+	"github.com/twmb/franz-go/pkg/sasl/scram"
+)
+
+var (
+	// TODO make the timeout configurable
+	connectionTimeout = time.Second * 10
+)
+
+// Config contains common configuration parameters.
+type Config struct {
+	// Servers is a list of Kafka bootstrap servers, which will be used to
+	// discover all the servers in a cluster.
+	Servers []string `json:"servers" validate:"required"`
+	// Topic is the Kafka topic.
+	Topic string `json:"topic" validate:"required"`
+
+	// ClientID is a unique identifier for client connections established by
+	// this connector.
+	ClientID string `json:"clientID" default:"conduit-connector-kafka"`
+
+	// -- TLS --
+
+	// ClientCert is the Kafka client's certificate.
+	ClientCert string `json:"clientCert"`
+	// ClientKey is the Kafka client's private key.
+	ClientKey string `json:"clientKey"`
+	// CACert is the Kafka broker's certificate.
+	CACert string `json:"caCert"`
+	// InsecureSkipVerify defines whether to validate the broker's certificate
+	// chain and host name. If 'true', accepts any certificate presented by the
+	// server and any host name in that certificate.
+	InsecureSkipVerify bool `json:"insecureSkipVerify"`
+
+	// -- SASL --
+
+	// SASLMechanism configures the connector to use SASL authentication. If
+	// empty, no authentication will be performed.
+	SASLMechanism string `json:"saslMechanism" validate:"inclusion=PLAIN|SCRAM-SHA-256|SCRAM-SHA-512"`
+	// SASLUsername sets up the username used with SASL authentication.
+	SASLUsername string `json:"saslUsername"`
+	// SASLPassword sets up the password used with SASL authentication.
+	SASLPassword string `json:"saslPassword"`
```

Comment:
Nitpick: we can combine these into a struct, which could then be more easily validated. Similar for TLS configs.

Reply:
Good idea!